File: BinarySearch/C++/367-Valid-Perfect-Square.cpp
class Solution {
public:
    bool isPerfectSquare(int num) {
        if(num == 0) return false;
        int left = 1, right = num;
        while(left < right){
            int mid = (left + (long long)right) >> 1;
            if(mid >= num / mid) right = mid;
            else left = mid + 1;
        }
        return left == sqrt(num);
    }
};

File: BinarySearch/Go/367-Valid-Perfect-Square.go
import "math"

func isPerfectSquare(num int) bool {
    if num == 0{
        return false;
    }

    left, right := 1, num
    for left < right{
        mid := (left + right) >> 1;
        if mid >= num / mid{
            right = mid
        }else{
            left = mid + 1
        }
    }
    return float64(left) == math.Sqrt(float64(num))
}